tests/TestAstar.py

```diff
@@ -1,42 +1,60 @@
-            X = X[:50, :]
-            dag_est, search_stats = bic_exact_search(X, search_method='astar')
-            print(dag_est)
-            print(search_stats)
+######################################### Test Notes ###########################################
+# All the benchmark results of loaded files (e.g. "./TestData/benchmark_returned_results/")    #
+# are obtained from the code of causal-learn as of commit                                      #
+# https://github.com/cmu-phil/causal-learn/commit/8badb41 (07-08-2022).                        #
+#                                                                                              #
+# We are not sure if the results are completely "correct" (reflect ground truth graph) or not. #
+# So if you find your tests failed, it means that your modified code is logically inconsistent #
+# with the code as of 8badb41, but not necessarily means that your code is "wrong".            #
+# If you are sure that your modification is "correct" (e.g. fixed some bugs in 8badb41),       #
+# please report it to us. We will then modify these benchmark results accordingly. Thanks :)   #
+######################################### Test Notes ###########################################
+
+
+BENCHMARK_TXTFILE_TO_MD5 = {
+    "./TestData/test_exact_search_simulated_linear_gaussian_data.txt": "1ec70464e4fc68c312adfb7143bd240b",
+    "./TestData/test_exact_search_simulated_linear_gaussian_CPDAG.txt": "52a6d3c5db269d5e212edcbb8283aca9",
+}
+# verify files integrity first
+for file_path, expected_MD5 in BENCHMARK_TXTFILE_TO_MD5.items():
+    with open(file_path, 'rb') as fin:
+        assert hashlib.md5(fin.read()).hexdigest() == expected_MD5,\
+            f'{file_path} is corrupted. Please download it again from https://github.com/cmu-phil/causal-learn/blob/8badb41/tests/TestData'
 
-    # example2
-    # for data with multi-variate dimensions, astar.
-    def test_multi_astar(self):
-        with open("example_data2.pk", 'rb') as example_data:
-            # example_data = load(open("example_data2.pk", 'rb'))
-            example_data = load(example_data)
-            Data_save = example_data['Data_save']
-            trial = 0
-            X = Data_save[trial]
-            X = X - np.tile(np.mean(X, axis=0), (X.shape[0], 1))
-            X = np.dot(X, np.diag(1 / np.std(X, axis=0)))
-            X = X[:50, :]
-            dag_est, search_stats = bic_exact_search(X, search_method='astar')
-            print(dag_est)
-            print(search_stats)
+
+class TestAstar(unittest.TestCase):
+    # Load data and run Astar with default parameters.
+    def test_astar_simulate_linear_gaussian_with_local_score_BIC(self):
+        print('Now start test_astar_simulate_linear_gaussian_with_local_score_BIC ...')
+        truth_CPDAG_matrix = np.loadtxt("./TestData/test_exact_search_simulated_linear_gaussian_CPDAG.txt")
+        data = np.loadtxt("./TestData/test_exact_search_simulated_linear_gaussian_data.txt")
```

Comment:
How do we generate this data? Do you mind adding the data generating script in this file as well (but comment out the code) --- this can ensure we know how the data is generated later.

Reply:
Thanks for the suggestion. I have added and commented the code to generate the data and ground truth.